### PR TITLE
Add support for `make_latest` option when creating or updating a release

### LINKED ISSUE #268: Support Explicitly Set the Latest Release
**Is your feature request related to a problem? Please describe.**
Explicitly set the release is the latest.
Refer https://github.blog/changelog/2022-10-21-explicitly-set-the-latest-release/

> Previously, a repository's latest release was the one created on the most recent date. In the event that multiple releases had the same date, the semantic version number broke the tie.

**Describe the solution you'd like**
add a option like `setAsLatest`, default to `true`

**Additional context**
Maybe need to wait rest api doc update

### PR DESCRIPTION
**Pull request by ericcornelissen**
Closes #268

Add a new input named `makeLatest` (chosen based on [the official docs](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release)) to configure whether the created or updated release should be marked as ["Latest"](https://github.blog/changelog/2022-10-21-explicitly-set-the-latest-release/) or not.

From the `make_latest` documentation:

> Specifies whether this release should be set as the latest release for the repository. Drafts and prereleases cannot be set as latest. Defaults to true for newly published releases. legacy specifies that the latest release should be determined based on the release creation date and higher semantic version.
>
> Default: `true`
>
> Can be one of: `true`, `false`, `legacy`

I'm unsure about the behavior for updating a release: in particular if the default for `makeLatest` is `true`, updating a release might unexpectedly make that release the latest release?

I'm also unsure whether or not this action's documentation should include a note that "Drafts and prereleases cannot be set as latest." It seems self-evident to me so I didn't include it.

### CODE REVIEW

**Comment by ericcornelissen** [comment]
Thanks for running the CI for this branch @ncipollo :slightly_smiling_face: I updated my contribution based on the CI logs:
1. Ran `yarn build` and included the resulting changes.
2. Added a test for the `makeLatest` input to the `CoreInputs` test suite to reach the coverage thresholds. Doing this made me realize my implementation was probably wrong, so changed the implementation as well.

File: src/Releases.ts [hunk 1]
```diff
@@ -26,7 +26,8 @@ export interface Releases {
         draft?: boolean,
         generateReleaseNotes?: boolean,
         name?: string,
-        prerelease?: boolean
+        prerelease?: boolean,
+        makeLatest?: boolean
```
Comment (ncipollo) [inline src/Releases.ts:30]: Nit: Could we move `makeLatest` up above name so this function is roughly 🔡?

File: src/Releases.ts [hunk 2]
```diff
@@ -88,7 +91,8 @@ export class GithubReleases implements Releases {
             prerelease: prerelease,
             repo: this.inputs.repo,
             target_commitish: commitHash,
-            tag_name: tag
+            tag_name: tag,
+            make_latest: makeLatest
```
Comment (ncipollo) [inline src/Releases.ts:95]: Hmm should this be a string? The documentation indicates this should be a string with 3 values: `true`, `false` & `legacy`.

File: action.yml [hunk 3]
```diff
@@ -47,6 +47,10 @@ inputs:
     description: 'Indicates if release notes should be automatically generated.'
     required: false
     default: 'false'
+  makeLatest:
+    description: 'Indicates if the release should be the "latest" release or not.'
+    required: false
+    default: 'true'
```
Comment (ncipollo) [inline action.yml:53]: I think the default should be `legacy` so the action behaves the same as it did before this change by default.

File: README.md [hunk 4]
```diff
@@ -25,6 +25,7 @@ This action will create a GitHub release and optionally upload an artifact to it
...
 | draft                      | Optionally marks this release as a draft release. Set to true to enable.                                                                                                                                                                            | false      | ""                   |
 | generateReleaseNotes       | Indicates if release notes should be automatically generated.                                                                                                                                                                                       | false      | false                |
+| makeLatest                 | Indicates if the release should be the "latest" release or not. If this is omitted the release will be made the latest release.                                                                                                                     | false      | true                 |
```
Comment (ncipollo) [inline README.md:28]: Same comment here - I think the default should be `legacy`

File: src/Inputs.ts [hunk 5]
```diff
@@ -136,6 +137,14 @@ export class CoreInputs implements Inputs {
         return generate == 'true'
     }
 
+    get makeLatest(): boolean {
```
Comment (ncipollo) [inline src/Inputs.ts:140]: I think this should be a string input so we can select true, false or legacy.

**Review by ncipollo — CHANGES_REQUESTED** [review]
Thanks for the PR!!🎉 

I had the following change requests:
- I think we probably want to make this input a string to match the API.
- I think we want to default the input to `legacy` so the default behavior is unchanged for folks.
- Had a minor nitpick about method argument ordering.

File: src/Releases.ts (hunk 2 above)
Comment (ericcornelissen) [reply src/Releases.ts]: I think you're right but it looks like `make_latest` is not yet included in the package that provides these types (which I believe is [`@octokit/plugin-rest-endpoint-methods`](https://www.npmjs.com/package/@octokit/plugin-rest-endpoint-methods)?). I tried re-installing [`@actions/github`](https://www.npmjs.com/package/@actions/github) locally, but I still didn't get type hints for this field (currently used is v5.13.0, after re-install I got v5.16.2, but latest is v6.7.0)...

File: src/Inputs.ts (hunk 5 above)
Comment (ericcornelissen) [reply src/Inputs.ts]: I updates this to return `undefined` if the option wasn't provided. Not sure if that's what you want or if it instead should return `"legacy"` :thinking:

Comment (ncipollo) [reply src/Inputs.ts]: Hmm, not sure - my guess would be undefined == legacy, so either would probably be sufficient. I can test with my playground repo and we can always change the default if it's wrong after.

**Pull request merged**